src/client/texturesource.cpp

```diff
@@ -521,7 +521,7 @@ video::SColor TextureSource::getTextureAverageColor(const std::string &name)
 	if (!image)
 		return {0, 0, 0, 0};
 
-	video::SColor c = ImageSource::getImageAverageColor(*image);
+	video::SColor c = imageAverageColor(image);
```

Comment:
> This function (getTextureAverageColor) should probably get a "pointless roundtrip through the gpu" fixme comment?

I have one in a different PR

> and getNormalTexture above is unused, could delete that (since this is already a random stuff PR, suggesting more random stuff to add :)

indeed, but this PR is big enough as-is